individual-files/Procedure.java

```diff
@@ -0,0 +1,38 @@
+import java.lang.Runtime;
+import java.io.*;
+
+public class Procedure extends HttpServlet
+{
+    public void doGet(HttpServletRequest req, HttpServletResponse res)
+        throws ServletException, IOException
+                        {
+        res.setContentType("text/html");
+        ServletOutputStream out = res.getOutputStream();
+        out.println("<HTML><HEAD><TITLE>Test</TITLE></HEAD><BODY><blockquote><pre>");
+        
+        String user = req.getParameter("user");
+        if(user != null) {
+            try {
+                 String[] args = { "/bin/sh", "-c", "finger " + user };
```

Comment:
This line is allowing a user defined string to be part of a command to be executed in a shell. This isn't safe at all. Some sanitization should be done to the "user" variable before it is used.